### ISSUE TITLE
Semibold fonts are not treated as bold

### ISSUE THREAD

**Issue by GrygrFlzr** [libass/libass#317]
When a semibold font is used and a bold variant is requested, libass will faux bold the semibold, whereas xy-vsfilter will treat the semibold as already bolded and use the font as is.

Note in this case I don't necessarily think libass's behavior is incorrect (semibold is a different weight from bold after all), but it *does* differ from xy-vsfilter, and I've been told to report any rendering differences here.

Reduced Case (1280x720):
`{\an2\pos(640,360)\fs80\fnSource Sans Pro Semibold\b1}(Bolded)\NThe quick brown fox jumps over the lazy dog`
`{\an8\pos(640,360)\fs80\fnSource Sans Pro Semibold\b0}The quick brown fox jumps over the lazy dog\N(Regular)`

Fonts are available [from GitHub releases](https://github.com/adobe-fonts/source-sans-pro/releases/tag/2.020R-ro%2F1.075R-it).
This is a distinct family from Source Sans Variable, which is the "latest" release on that repo. The TTF versions were used.

Explicitly requesting Source Sans Pro Semibold will make it distinct from the rest of the Source Sans Pro family so neither xy-vsfilter nor libass will pick up Source Sans Pro Bold for the bold variant even when it is installed/attached.

Hover comparison available [here](https://diff.pics/GCkex9JuneU6/1), alternatively the render outputs are embedded below:
![image](https://user-images.githubusercontent.com/1460917/47489095-cb349800-d813-11e8-8802-bcc3d3a7e009.png)
![image](https://user-images.githubusercontent.com/1460917/47489070-bd7f1280-d813-11e8-8d9d-e0b61e6faf38.png)

**Comment by rcombs** [comment]
I believe this is fixed in 8a59cd874c11bc7c8f4906f0f98360a5b021110f. The VSFilter behavior is to only faux-bold if the weight difference is over 150.

**Comment by GrygrFlzr** [comment]
Just tested with the Windows mpv build (git) from an hour ago, which I think based on the build script *should* use the latest version of libass. It does not seem to have changed behavior, but I'm not entirely sure how to check if it's actually running the latest version of libass or not.

**Comment by GrygrFlzr** [comment]
I ran it again with verbose logs and found something interesting:
```
[sub/ass] fontselect: (Source Sans Pro Semibold, 400, 0) -> SourceSansPro-Semibold, 0, SourceSansPro-Semibold
[sub/ass] fontselect: (Source Sans Pro Semibold, 700, 0) -> SourceSansPro-Semibold, 0, SourceSansPro-Semibold
```
So it is treating the original semibold as weight 400 instead of 600. I checked the font in Font Forge and while it is correctly marked as weight 600 and Semi/Demibold, it has the Style Map "None" attached, which may be relevant?

**Comment by rcombs** [comment]
8a59cd8 isn't in master yet.
Getting correct weights should be fixed in f2439efb2a30e94a840082d105cce1316b7eeb6d, which also isn't in master.

**Comment by astiob** [comment]
Fixed in ac75f67697845efc1b0a15693c96dcc9e0f71518 and 6591a60a3d720f576352f8b4ef08362673d35cd0.
